packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -224,26 +257,56 @@ export default function MainFeedLayout({
         emptyScreen: <SearchEmptyScreen />,
       };
     }
-    if (!query.query) {
+
+    if (!config.query) {
       return null;
     }
 
     const getVariables = () => {
-      if (isUpvoted) {
-        return { ...query.variables, period: periods[selectedPeriod].value };
+      if (isUpvoted || feedWithDateRange.includes(tab)) {
+        return { ...config.variables, period: periods[selectedPeriod].value };
+      }
+
+      if (isAnyExplore) {
+        const laptopValue =
+          tab === ExploreTabs.ByDate || isExploreLatest ? 1 : 0;
+        const finalAlgo = isLaptop ? laptopValue : selectedAlgo;
+        return {
+          ...config.variables,
+          ranking: algorithms[finalAlgo].value,
+        };
       }
 
       if (isSortableFeed) {
         return {
-          ...query.variables,
+          ...config.variables,
           ranking: algorithms[selectedAlgo].value,
         };
       }
 
-      return query.variables;
+      return config.variables;
     };
 
     const variables = getVariables();
+    const classes = isAnyExplore && {
+      wrapper: '!min-h-0',
+      emptyScreen: 'mt-8 h-auto',
+    };
+    const mobileExploreActions = !isLaptop &&
+      (isExplorePopular || isExploreLatest) && (
+        <Dropdown
+          className={{ container: 'mx-4 mb-2 !w-56' }}
+          selectedIndex={selectedAlgo}
+          options={algorithmsList}
+          onChange={(_, index) => setSelectedAlgo(index)}
+        />
+      );
+    const controlFeedActions = feedWithActions && (
+      <SearchControlHeader
+        algoState={[selectedAlgo, setSelectedAlgo]}
+        feedName={feedName}
+      />
+    );
```

Comment:
Still isn't this basically the exact same component?

Reply:
Unfortunately, we don't render this component on mobile which is exactly the opposite of what we want - which is to render on mobile. Also, there are certain elements from there that we don't to render, like the streak-related ones.

Reply:
Ah ok :)